tests/test_temporal.py

```diff
@@ -831,7 +831,7 @@ def test_weighted_monthly_averages_with_masked_data(self):
         expected = expected.drop_dims("time")
         expected["ts"] = xr.DataArray(
             name="ts",
-            data=np.array([[[2.0]], [[0.0]], [[1.0]], [[1.0]], [[2.0]]]),
+            data=np.array([[[2.0]], [[np.nan]], [[1.0]], [[1.0]], [[2.0]]]),
```

Comment:
The value that was `0.0` is taking a monthly average for a month with one data point, which is `np.nan`. The average of one `np.nan` value should be `np.nan`.